internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -94,37 +96,137 @@ func (c *cloudeventProcessor) processMsg(ctx context.Context, msg *service.Messa
 		processors.SetError(msg, processorName, "failed to get message as bytes", err)
 		return service.MessageBatch{msg}
 	}
-	source, ok := msg.MetaGet(httpinputserver.DIMOConnectionIdKey)
+	source, ok := msg.MetaGet(httpinputserver.DIMOCloudEventSource)
 	if !ok {
 		processors.SetError(msg, processorName, "failed to get source from message metadata", err)
 		return service.MessageBatch{msg}
 	}
-	hdrs, eventData, err := modules.ConvertToCloudEvents(ctx, source, msgBytes)
-	if err != nil {
-		// Try to unmarshal convert errors
-		data, marshalErr := json.Marshal(err)
-		if marshalErr == nil {
-			msg.SetBytes(data)
-		}
-		processors.SetError(msg, processorName, "failed to convert to cloud event", err)
-		return service.MessageBatch{msg}
-	}
-	if len(hdrs) == 0 {
-		processors.SetError(msg, processorName, "no cloud events headers returned", nil)
+
+	contentType, ok := msg.MetaGet(processors.MessageContentKey)
+	if !ok {
+		processors.SetError(msg, processorName, "failed to get content type from message metadata", err)
 		return service.MessageBatch{msg}
 	}
-	if len(eventData) == 0 {
-		// If the module chooses not to return data, use the original message will be used
-		eventData = msgBytes
+
+	var hdrs []cloudevent.CloudEventHeader
+	var eventData []byte
+	switch contentType {
+	case httpinputserver.ConnectionContent:
+		hdrs, eventData, err = modules.ConvertToCloudEvents(ctx, source, msgBytes)
+		if err != nil {
+			// Try to unmarshal convert errors
+			data, marshalErr := json.Marshal(err)
+			if marshalErr == nil {
+				msg.SetBytes(data)
+			}
+			processors.SetError(msg, processorName, "failed to convert to cloud event", err)
+			return service.MessageBatch{msg}
+		}
+		if len(hdrs) == 0 {
+			processors.SetError(msg, processorName, "no cloud events headers returned", nil)
+			return service.MessageBatch{msg}
+		}
+		if len(eventData) == 0 {
+			// If the module chooses not to return data, use the original message will be used
+			eventData = msgBytes
+		}
+	case httpinputserver.AttestationContent:
+		var event cloudevent.CloudEvent[json.RawMessage]
+		err = json.Unmarshal(msgBytes, &event)
+		if err != nil {
+			processors.SetError(msg, processorName, "failed to unmarshal attestation cloud event", err)
+			return service.MessageBatch{msg}
+		}
+
+		currentTimestamp := time.Now().UTC()
+		if !(currentTimestamp.Sub(event.Time).Minutes() <= 5 && event.Time.Sub(currentTimestamp).Minutes() <= 5) {
+			processors.SetError(msg, processorName, fmt.Sprintf("event timestamp %+v exceeds valid range", event.Time), nil)
+			return service.MessageBatch{msg}
+		}
+
+		if event.ID == "" {
+			event.ID = ksuid.New().String()
+		}
+
+		if event.DataContentType == "" {
+			event.DataContentType = "application/json"
+		}
+
+		if event.SpecVersion == "" {
+			event.SpecVersion = "1.0"
+		}
```

Comment:
I think these are valid defaults for all cloud events so we can add them to the setDefaults function which is called in createEventMessage